src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlDateAddTranslator.cs

```diff
@@ -29,39 +42,65 @@ public class NpgsqlDateAddTranslator : IMethodCallTranslator
         };
 
         /// <summary>
-        ///     Translates the given method call expression.
+        /// The backend process to target.
+        /// </summary>
+        readonly Backend _backend;
+
+        /// <summary>
+        /// The backend version to target.
+        /// </summary>
+        [NotNull] readonly Version _compatibility;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NpgsqlDateAddTranslator"/> class.
         /// </summary>
-        /// <param name="methodCallExpression">The method call expression.</param>
-        /// <returns>
-        ///     A SQL expression representing the translated MethodCallExpression.
-        /// </returns>
+        /// <param name="backend">The backend process to target.</param>
+        /// <param name="compatibility">The backend version to target.</param>
+        public NpgsqlDateAddTranslator(Backend backend, [NotNull] Version compatibility)
+        {
+            _backend = backend;
+            _compatibility = Check.NotNull(compatibility, nameof(compatibility));
+        }
+
+        /// <inheritdoc />
         public virtual Expression Translate(MethodCallExpression methodCallExpression)
         {
-            if (!_methodInfoDatePartMapping.TryGetValue(methodCallExpression.Method, out var datePart))
+            // This translation is only supported for PostgreSQL 9.4 or higher.
+            if (_backend != Backend.PostgreSQL || _compatibility < MinimumSupportedVersion)
```

Comment:
I'll remove the backend type check for now. But this raises the question of which version do we mean for `BackendVersion` to represent? 

Should it be the _actual_ backend's version (e.g. `type: Redshift, version: Redshift-X.X`) or the _equivalent_ PostgreSQL version (e.g. `type: Redshift, version: PostgreSQL-9.4`)?

Reply:
That's a good question, I can see arguments both ways... I'm not sure we have enough actual information on this.

Redshift forked from PostgreSQL around version 8.0.2, and has not progressed in terms of "PostgreSQL versions" - AFAIK there are no post-8.0.2 PostgreSQL language features that have been ported over to Redshift. But in any case running EF Core over Redshift may very well be much too much effort, and not necessarily a great fit. I wouldn't worry about it too much.

For the others (CockroachDB, CrateDB) I simply don't know... In the absence of information it may be safer to simply have a `PostgresVersion` (rather than `BackendVersion`, less ambiguous), and possibly have the optional `BackendType` determine a `PostgresVersion`... So that if `BackendType` is set to be Redshift, we'd automatically set `PostgresVersion` to be 8.0.2 (in addition to making some other, truly Redshift-specific behavior modifications). But this is really largely theoretical at this point.

Makes sense?